src/blob/index/simple.rs

```diff
@@ -138,14 +152,19 @@ impl Simple {
             .await
             .with_context(|| format!("file open failed {:?}", self.name.to_path()))?;
         let size = file.write_append(&buf).await?;
+        file.write_append(&MAGIC_NUMBER.to_be_bytes()).await?;
```

Comment:
I suggest to write it to the header, not to the end of the file. With the current approach, it will be potentially more difficult to extend the index format